### PR TITLE
Using update options to create or incrementally update entities

### PR DESCRIPTION
**Pull request by gs-gunjan**
using update options to create or incrementally update entities

### CODE REVIEW

**Comment by MauricioUyaguari** [comment]
@gs-gunjan can we add a test here with entities with "." in the field name in the content.

File: legend-depot-store-mongo/src/main/java/org/finos/legend/depot/store/mongo/entities/EntitiesMongo.java [hunk 1]
```diff
@@ -143,9 +149,22 @@ protected void validateNewData(StoredEntity entity)
     @Override
     public List<StoredEntity> createOrUpdate(List<StoredEntity> versionedEntities)
     {
-        List<StoredEntity> entities = new ArrayList<>();
-        versionedEntities.forEach(item -> entities.add(createOrUpdate(item)));
-        return entities;
+        versionedEntities.forEach(item ->
```
[the remark this answers is not in the document]
Comment (gs-gunjan) [reply legend-depot-store-mongo/src/main/java/org/finos/legend/depot/store/mongo/entities/EntitiesMongo.java:152]: with create or update we use the object mapper for serialization and there seems to be a little bit of difference against what mongo java directly converts to

**Review by elopezcastro — APPROVED** [review]

File: legend-depot-store-mongo/src/test/resources/data/versioned-entity-dot-character.json [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    "groupId": "examples.metadata",
+    "artifactId": "test",
+    "versionedEntity": false,
+    "entity": {
+      "path": "examples::metadata::test::TestProfile",
+      "classifierPath": "meta::pure::metamodel::extension::Profile",
+      "content": {
+        "_type": "profile",
+        "name": "dataProfile.profile",
+        "package": "examples::metadata::test"
```
Comment (MauricioUyaguari) [inline legend-depot-store-mongo/src/test/resources/data/versioned-entity-dot-character.json:13]: Can we add here a key with ".". We have a value with "." but i think there are different rules for keys and values.

Comment (gs-gunjan) [reply legend-depot-store-mongo/src/test/resources/data/versioned-entity-dot-character.json:13]: The error we found was not in key, but the value itself, the object mapper serialization made it a key

**Pull request merged**